dags/roger/core.py

```diff
@@ -725,20 +726,29 @@ def write_items_to_redis(self, items):
                 pipeline.set(key, json.dumps(items[key]))
             pipeline.execute()
 
+    def batch_keys(self, batch_size):
+        from itertools import zip_longest
+        keyspace = self.redis_conn.info('keyspace').get('db1', {}) .get('keys', 1)
```

Comment:
Two questions:
1. This hard-codes the DB ID, don't we have that info in an environment variable somewhere?
2. Should there be some exception or other log message if it can't connect or get the keys, instead of defaulting to 1?

Reply:
1. we are using db1 (here https://github.com/helxplatform/roger/blob/effd6021f42445918074eaa617929e6bb6f97def/dags/roger/core.py#L412) maybe we should make that an env var and resolve as such. 
2. if it can't connect it would raise an error, but if keyspace['db1'] doesn't have any value it just means that its empty , so by defaulting to one we the next call to get all keys would have empty list. we can return empty list if keyspace[db1] is empty though

Reply:
1. I have added a new config under kgx section in the config.yaml `merge_db_id` that would be used to choose a redis db id . 
2. added a better way of dealing with empty keyspace